semgrep-core/src/engine/Unit_engine.ml

```diff
@@ -284,10 +284,9 @@ let regression_tests_for_lang ~with_caching files lang =
              Common.save_excursion Flag_semgrep.with_opt_cache with_caching
                (fun () ->
                  Match_patterns.check
-                   ~hook:(fun _env matched_tokens ->
+                   ~hook:(fun { Pattern_match.tokens = (lazy xs); _ } ->
```

Comment:
what? I didn't know you could use 'lazy' in a pattern :) nice.

Reply:
I think this is actually considered more idiomatic than `Lazy.force` and I use it where I can!